azure/services/natgateways/spec.go

```diff
@@ -91,8 +91,8 @@ func (s *NatGatewaySpec) Parameters(ctx context.Context, existing interface{}) (
 }
 
 func hasPublicIP(natGateway armnetwork.NatGateway, publicIPName string) bool {
-	// We must have a non-nil, non-"empty" PublicIPAddresses
-	if !(natGateway.Properties.PublicIPAddresses != nil && len(natGateway.Properties.PublicIPAddresses) > 0) {
+	// PublicIPAddresses must not be nil or empty.
+	if len(natGateway.Properties.PublicIPAddresses) == 0 {
 		return false
 	}
```

Comment:
I suppose we could get rid of this entirely since an empty slice will basically skip over the for loop and return false anyway.